model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -0,0 +1,243 @@
+            node.begin_mask[ellipsis_start] = 0
+            node.end_mask[ellipsis_start] = 0
+            node.ellipsis_mask[ellipsis_start] = 0
+            insertation_start_idx = ellipsis_start + 1
+
+            StridedSliceNormalizer.unroll_ellipsis_for_inputs(graph, node, ellipsis_start, num_insertions)
+        elif num_insertions > 0:
+            insertation_start_idx = slice_rank  # insert blank values to mask ends
+            StridedSliceNormalizer.extend_inputs(node, num_insertions)
+
+        if num_insertions > 0:
+            # insert blank values for ellipsis unrolling and extending
+            for mask_name in StridedSlice.get_mask_names():
+                node[mask_name] = np.insert(node[mask_name], insertation_start_idx, [0] * num_insertions).astype(int)
+
+    @staticmethod
+    def unroll_ellipsis_for_inputs(graph: Graph, node: Node, ellipsis_start: int, num_ellipsis_ext: int):
+        node_name = node.soft_get('name', node.id)
+        for i, slice_name in enumerate(('begin', 'end', 'strides')):
+            i += 1
+            placeholder_arr = np.zeros(num_ellipsis_ext) if i != 3 else np.ones(num_ellipsis_ext)
+            placeholder_node = Const(graph, {'name': node_name + '/const_to_unroll_{}_ellipsis'.format(slice_name),
+                                             'value': int64_array(placeholder_arr)}).create_node()
+
+            concat_in_ports_count = 3 if ellipsis_start != 0 else 2
+            concat = Concat(graph, {'axis': 0, 'name': node_name + '/concat_{}'.format(slice_name),
+                                    'in_ports_count': concat_in_ports_count}).create_node()
+
+            if ellipsis_start != 0:
+                split = create_op_with_const_inputs(graph, VariadicSplit, {1: int64_array(0),
+                                                                           2: int64_array([ellipsis_start, -1])},
+                                                    {'name': node_name + '/split_for_{}_ellipsis'.format(slice_name),
+                                                     'out_ports_count': 2})
+                node.in_port(i).get_connection().set_destination(split.in_port(0))
+
+                concat.in_port(0).connect(split.out_port(0))
+                concat.in_port(1).connect(placeholder_node.out_port(0))
+                concat.in_port(2).connect(split.out_port(1))
+            else:
+                concat.in_port(0).connect(placeholder_node.out_port(0))
+                node.in_port(i).get_connection().set_destination(concat.in_port(1))
+
+            concat.out_port(0).get_connection().set_destination(node.in_port(i))
+
+    @staticmethod
+    def extend_inputs(node: Node, num_insertations: int):
+        graph = node.graph
+        node_name = node.soft_get('name', node.id)
+
+        for i, slice_name in enumerate(('begin', 'end', 'strides')):
+            i += 1
+            if node.in_port(3).disconnected():
```

Comment:
should we move it to the top of the method?

Reply:
I am not sure about that.

I left it here but replaced `break` with `continue`. Now it looks less confusing and more flexible, if for some reason `for` will be changed -> `for i, input_name in [(1, 'begin'), (3, 'strides'), (2, 'end')]` current solution will still work.